testing/internal/e2e/helpers.go

```diff
@@ -0,0 +1,53 @@
+package e2e
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+// CommandResult encapsulates the output running an external command
+type CommandResult struct {
+	Stdout   []byte
+	Stderr   []byte
+	ExitCode int
+}
+
+const envToCheck = "E2E_PASSWORD_AUTH_METHOD_ID"
+
+// RunCommand executes external commands on the system. Returns the results
+// of running the provided command.
+func RunCommand(name string, args ...string) (*CommandResult, error) {
+	var outbuf, errbuf bytes.Buffer
+
+	cmd := exec.Command(name, args...)
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+
+	err := cmd.Run()
+
+	var ee *exec.ExitError
+	var exitCode int
+	if errors.As(err, &ee) {
+		exitCode = ee.ExitCode()
+	}
+
+	return &CommandResult{
+		Stdout:   outbuf.Bytes(),
+		Stderr:   errbuf.Bytes(),
+		ExitCode: exitCode,
+	}, err
+}
+
+// TestInit is a helper method to set up individual tests.
+func TestInit(t *testing.T) {
+	if _, ok := os.LookupEnv(envToCheck); !ok {
+		t.Skip(fmt.Sprintf(
+			"Skipping test because environment variable '%s' is not set. This is needed for e2e tests.",
+			envToCheck,
+		))
+	}
+}
```

Comment:
I think this function might harm more than it helps. My biggest concern is that when calling this function in a test it obscures the fact that we might entirely skip the test, especially since `t.Skip` already messes with control flow magically. Could we inline this function wherever we're using it, instead of using this helper?

Reply:
Curious, what if we renamed the method to something like `CheckIfTestShouldBeSkipped`?

Reply:
`MaybeSkipTest`?

Reply:
Works for me.